### PR TITLE
[stable7] Only rescan versions once in trashbin

### PR DESCRIPTION
**Pull request by PVince81**
Whenever versions need to be rescanned, only do it once per PHP request.
Happens whenever multiple files need to be expired.

Backport of https://github.com/owncloud/core/pull/14640 to stable7.

@karlitschek @cmonteroluque @DeepDiver1975 

@karlitschek please approve, if applicable

### CODE REVIEW

**Comment by karlitschek** [comment]
Thanks a lot @PVince81 Please backport :+1:   :-)

**Comment by DeepDiver1975** [comment]
:+1:

**Comment by karlitschek** [comment]
Great. Please backport :+1:

**Comment by PVince81** [comment]
Ahem, this WAS the backport.

**Comment by karlitschek** [comment]
Hehe. Then I rephrase this to "Please merge this backport"   ;-) ;-)
